evap/staff/views.py

```
@@ -479,17 +426,72 @@ def course_comments(request, semester_id, course_id):
 
-    textanswers = course.textanswer_set.filter(checked=True)
-
-    textanswers_by_question = []
-    for question_id in textanswers.values_list("question", flat=True).distinct():
-        textanswers_by_question.append((Question.objects.get(id=question_id), textanswers.filter(question=question_id)))
+    course_sections = []
+    contributor_sections = []
+    for questionnaire, contribution in questionnaires_and_contributions(course):
+        text_results = []
+        for question in questionnaire.question_set.all():
+            if question.is_text_question:
+                answers = get_filtered_answers(course, contribution, question, exclude_text_answer_states=[])
+                if answers:
+                    text_results.append(TextResult(question=question, answers=answers))
+        if not text_results:
+            continue
+        if contribution.is_general:
+            course_sections.append(CommentSection(questionnaire, contribution.contributor, False, text_results))
+        else:
+            contributor_sections.append(CommentSection(questionnaire, contribution.contributor, contribution.responsible, text_results))
 
-    template_data = dict(semester=semester, course=course, textanswers_by_question=textanswers_by_question)
+    template_data = dict(semester=semester, course=course, course_sections=course_sections, contributor_sections=contributor_sections)
     return render(request, "staff_course_comments.html", template_data)
 
 
 @staff_required
+def course_comments_update_publish(request):
+    comment_id = request.POST["id"]
+    action = request.POST["action"]
+    course_id = request.POST["course_id"]
+
+    course = Course.objects.get(pk=course_id)
+    answer = TextAnswer.objects.get(pk=comment_id)
+
+    if action == 'publish':
+        answer.publish()
+    elif action == 'private':
+        answer.make_private()
+    elif action == 'hide':
+        answer.hide()
+    elif action == 'unreview':
+        answer.unreview()
```

could you make this method fail loudly in an else here?